src/shared/components/home/home.tsx

```diff
@@ -650,6 +661,8 @@ export class Home extends Component<any, HomeState> {
 
     if (dataType === DataType.Post) {
       switch (this.state.postsRes?.state) {
+        case "empty":
```

Comment:
Seeing this makes me think we should default to setting the responses to `{ state: "loading" }` instead of `{ state: "empty" }`. That way, it will display the loading spinner when fetching posts instead of a big empty div.

Reply:
When loading, it should immediately switch from `"empty"` to `"loading"` state as rendering that state is near instant. The `"empty"` state is only for scrolling while the `"success"` state renders, which is fast but slow enough to stop proper scrolling and just end up where the bottom of the empty page is. If we wait for the page to render, the transition isn't as smooth with a jerky snap when the content finishes loading. It's something the user should never really see, and more of a hack to make the scrolling work as intended.